src/ui/components/ObjectView.vue

```diff
@@ -148,9 +176,29 @@ export default {
                 this.currentObject = mutatedObject;
             });
 
+            this.unlistenStyles = this.openmct.objects.observe(this.currentObject, 'configuration.conditionalStyle', (mutatedObject) => {
+                //Updating conditional styles in the inspector view will trigger this so that the changes are reflected immediately
+                this.initConditionalStyles();
+            });
+
             this.viewKey = viewKey;
+
+            this.initConditionalStyles();
+
             this.updateView(immediatelySelect);
         },
+        initConditionalStyles() {
+            if (this.styleRuleManager) {
+                this.styleRuleManager.destroy();
+                this.styleRuleManager.off('conditionalStyleUpdated', this.updateStyle.bind(this));
+                delete this.styleRuleManager;
+            }
+
+            if (this.currentObject.configuration && this.currentObject.configuration.conditionalStyle) {
+                this.styleRuleManager = new StyleRuleManager(this.currentObject, this.openmct);
```

Comment:
Would it simplify things if the StyleRuleManager could be updated with a new style definition when it changes? Rather than throwing away the StyleRuleManager, and creating a new one each time? Might reduce some of the boilerplate around listener management etc. This would involved refactoring the StyleRuleManager to just accept style rules (the conditionalStyle attribute) rather than domain objects, which might also improve Class cohesion?

Reply:
You're right! StyleRuleManager only uses the domainObject to get the conditionalStyle. I'll make this change.

Reply:
Done. This was a little more complex than I initially thought, but it works now. :)